lib/processing.py

```diff
@@ -0,0 +1,490 @@
+    """
+    # Convert word to lower case to handle case insensitivity
+    word = word.lower()
+    # List that represents only the characters of the bounding boxes
+    text_extracted = []
+    # List of the bounding boxes represented by each character, including only the coordinates of the bounding boxes
+    total_data = []
+    for box in bounding_boxes.split('\n'):
+        if box is not None:
+            text_extracted.append(box.split(' ')[0].lower())
+            total_data.append(box.split(' ')[1:])
+
+    letters_str = ''.join(text_extracted)
+    start_index = letters_str.find(word)
+    if start_index == -1:
+        return "Sequence not found"
+    # The end index is the start index plus the length of the word
+    end_index = start_index + len(word)
+    
+    list_bb_word = total_data[start_index:end_index]
+    return list_bb_word
+
+# convert a file in png the same format to extract data
+def convert_to_png(file_path):
+    """
+    Converts a file to PNG format.
+
+    Parameters:
+    - file_path (str): The path to the file.
+
+    Returns:
+    - new_path (str): The path to the converted PNG file.
+    - text (None): None.
+
+    Raises:
+    - None.
+    """
+    img = Image.open(file_path)
+    new_path = file_path.rsplit('.', 1)[0] + '.png'
+    img.save(new_path, format='PNG')
+    text = None
+    return new_path, text
+
+# Extract text from a PDF file using PyMuPDF and convert it to images if the outcome of PDF is empty to extract text using tesseract OCR
+def handle_pdf(file_path):
+    """
+    Handles a PDF file by extracting text from it.
+
+    Parameters:
+    - file_path (str): The path to the PDF file.
+
+    Returns:
+    - image_paths (list): A list of paths to the extracted images if the PDF is empty.
+    - text (str): The extracted text from the PDF if it is not empty.
+    """
+    # Extract text from PDF
+    doc = fitz.open(file_path)
+    text = ''
+    for page in doc:
+        text += page.get_text()
+
+    if len(text) == 0:
+        # Convert PDF to image
+        images = convert_from_path(file_path)
+        image_paths = []
+        base_path = file_path.rsplit('.', 1)[0]
+        for i, image in enumerate(images):
+            image_path = f"{base_path}_page_{i}.png"
+            image.save(image_path, "PNG")
+            image_paths.append(image_path)
+
+        return image_paths, text
+    else:
+        return None, text
+
+# Process the file to extract the text and bounding boxes
+def process_file(file_path, name, lastname, config, text=None):
+    highest_name_simitality = 0
+    highest_lastname_simitality = 0
+    if text:    
+        answer = get_answer_llm(config, text, 'prompt_template_1')
+        dictionary = convert_str_to_dict(answer)
+        name_similarity = calculate_similarity(dictionary["name"], name)
+        lastname_similarity = calculate_similarity(dictionary["last_name"], lastname)
+
+        return {"Name_extracted": dictionary["name"], 
```

Comment:
Same here for dataclassess, it would be better to create something like


```python
from pydantic import BaseModel

class ExtractionResponse(BaseModel):
      last_name_extracted: str
      fuzzy_matching_name: int | float .....

```